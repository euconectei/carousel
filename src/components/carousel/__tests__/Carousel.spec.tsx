import { fireEvent, render, screen } from "@testing-library/react";
import { Carousel } from "..";

const values = [
  {
    value: "/images/first-slide.jpg",
    type: "image",
  },
];

describe("Testing Carousel Component", () => {
  it("should create a carousel component", () => {
    const { container } = render(<Carousel items={values} />);
    // expect(container).toMatchSnapshot();
    expect(true).toBe(true);
  });
});

describe("Testing Carousel Component with controls", () => {
  it("should not render controls if not passed", async () => {
    render(<Carousel items={values} />);
    const buttons = await screen.queryAllByRole("button");
    expect(buttons.length).toBe(0);
  });

  it("should render controls if passed", async () => {
    render(<Carousel items={values} showControls />);
    const buttons = await screen.queryAllByRole("button");
    expect(buttons.length).toBe(2);
    buttons.forEach((button) => {
      expect(button.className.match(/(carousel-button-)\w+/g)).toBeTruthy();
    });
  });
});

describe("Testing Carousel Component with indicators", () => {
  it("should not render indicators if not passed", async () => {
    render(<Carousel items={values} />);
    const buttons = screen.queryAllByRole("button");
    expect(buttons.length).toBe(0);
  });

  it("should render indicators if passed", async () => {
    render(<Carousel items={values} showIndicators />);
    const buttons = screen.queryAllByRole("button");
    expect(buttons.length).toBe(values.length);
  });

  it("should change the active item on click", async () => {
    render(<Carousel items={values} showIndicators />);
    const buttons = await screen.findAllByRole("button");

    fireEvent.click(buttons[0]);

    const itemsIndicators = await screen.findAllByRole("listitem");
    expect(itemsIndicators[0].classList.contains("active")).toBeTruthy();
  });
});
