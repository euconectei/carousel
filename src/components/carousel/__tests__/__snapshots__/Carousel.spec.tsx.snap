// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component should create a carousel component 1`] = `
<body>
  <div>
    <div
      class="carousel"
    >
      <div
        class="carousel-wrapper"
      >
        <div
          class="carousel-container"
        >
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item"
              style="min-width: 0px;"
            >
              <div
                class="carousel-item-image"
              >
                <img
                  src="/images/first-slide.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Testing Carousel Component should create a carousel component 1`] = `
<body>
  <div>
    <div
      class="carousel"
    >
      <div
        class="carousel-wrapper"
      >
        <div
          class="carousel-container"
        >
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item"
              style="min-width: 0px;"
            >
              <div
                class="carousel-item-image"
              >
                <img
                  src="/images/first-slide.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
